### PR TITLE
Added a function getFOVAngle to Vec3.h and Vec3.cpp

### PR DESCRIPTION
**Pull request by amantley**
Added a function getFOVAngle() to Vec3.h and Vec3.cpp.
This function aids with finding whether an object is in the field of view.
This will extend the scripting API
-Angus Antley

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can one of the admins verify this patch?

**Comment by sethalves** [comment]
test this please

**Comment by sethalves** [comment]
gustavo what's wrong

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3129/HighFidelity-Beta-PR11322-e8860f5c82d49fb3cfe7f316cfeeea91487a2479.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3129/HighFidelity-Beta-PR11322-e8860f5c82d49fb3cfe7f316cfeeea91487a2479.exe

File: libraries/script-engine/src/Vec3.cpp [hunk 1]
```diff
@@ -93,3 +93,16 @@ float Vec3::getAngle(const glm::vec3& v1, const glm::vec3& v2) {
...
+    if (glm::abs(length(userObjectAxis)) < EPSILON || glm::abs(length(lookAt)) < EPSILON) {
+        //we have an error.
+        return -1;
+    }
+    else {
+        //return the angle in radians.
+        //the angle range is from 0, directly in front, to pi radians, directly behind.
+        return glm::acos(glm::dot(glm::normalize(lookAt), glm::normalize(userObjectAxis)));
+    }
+}
```
Comment (AndrewMeadows) [inline libraries/script-engine/src/Vec3.cpp]: While `Vec3::getFOVAngle()` is pure `Vec3` math it isn't fundamental: it is a specialization of the `Vec3::getAngle()` method.  Consider this bit of JS that computes the `fov` angle of an object relative to the `Camera`:
```
var lookat = Quat.getForward(Camera.getOrientation());
var posUser = Camera.getPosition();
var posObject = properties.position;
var fov = Vec3.getFOVAngle(lookAt, posUser, posObject);
```
This JS code does the same thing with existing `Vec3` API:
```
var lookat = Quat.getForward(Camera.getOrientation());
var relativePosObject = Vec3.subtract(properties.position, Camera.getPosition());
var fov = Vec3.getAngle(lookAt, relativePosObject);
```

**Comment by amantley** [comment]
Hi Andrew,

  I agree that this function is just a luxury.

  I like it mainly because it means I won't make an error of finding the
relative position of the object( i.e. swapping camera.position and
properties.position in Vec3.subtract)

  Angus

**Comment by ZappoMan** [comment]
Angus, can you provide some examples of other APIs that include this function so we can evaluate how common it is to include this utility as a helper.

Also, the code doesn't match the coding standard for brace placement, and need to be repaired.

Finally, we'd need a testplan/unit test for this as well.

**Comment by themelissabrown** [comment]
@amantley can you provide examples as per Zappoman's request above? Or should we close this PR?

**Comment by amantley** [comment]
Hi Melissa,

I wasn't able to find anything to directly match this function, as a method
of the vector class.

I did find an SDK for a game that had a function like this.
isinfieldofview(vec position)
And Unreal has a graphical AI package that supports input of an object and
output of peripheral vision angle.

https://wiki.alliedmods.net/CDODPlayer_Offset_List_(Day_of_Defeat:_Source)

<https://wiki.alliedmods.net/CDODPlayer_Offset_List_%28Day_of_Defeat:_Source%29>
https://docs.unrealengine.com/latest/INT/BlueprintAPI/AI/Components/PawnSensing/GetPeripheralVisionAngle/index.html
<https://wiki.alliedmods.net/CDODPlayer_Offset_List_%28Day_of_Defeat:_Source%29>


​I found this function useful for implementing npc gaze b​ehaviour,  so
maybe it could be a method of an npc entity/avatar rather than vec3?


Angus

**Pull request closed without merging**